spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/integration/IntegrationAutoConfiguration.java

```diff
@@ -39,19 +47,49 @@
 @Configuration
 @ConditionalOnClass(EnableIntegration.class)
 @AutoConfigureAfter(JmxAutoConfiguration.class)
+@EnableConfigurationProperties(IntegrationProperties.class)
 public class IntegrationAutoConfiguration {
 
 	@Configuration
 	@EnableIntegration
 	protected static class IntegrationConfiguration {
+
+		@Bean(name = IntegrationContextUtils.INTEGRATION_GLOBAL_PROPERTIES_BEAN_NAME)
+		@ConditionalOnMissingBean(name = IntegrationContextUtils.INTEGRATION_GLOBAL_PROPERTIES_BEAN_NAME)
+		public Properties integrationGlobalProperties(IntegrationProperties integrationProperties) {
+			Properties properties = new Properties();
+			properties.setProperty(org.springframework.integration.context.IntegrationProperties.CHANNELS_AUTOCREATE,
+					"" + integrationProperties.getChannels().isAutoCreate());
+			properties.setProperty(org.springframework.integration.context.IntegrationProperties.CHANNELS_MAX_UNICAST_SUBSCRIBERS,
+					Integer.toString(integrationProperties.getChannels().getMaxUnicastSubscribers()));
+			properties.setProperty(org.springframework.integration.context.IntegrationProperties.CHANNELS_MAX_BROADCAST_SUBSCRIBERS,
+					Integer.toString(integrationProperties.getChannels().getMaxBroadcastSubscribers()));
+			properties.setProperty(org.springframework.integration.context.IntegrationProperties.TASK_SCHEDULER_POOL_SIZE,
+					Integer.toString(integrationProperties.getTaskSchedulerPoolSize()));
+			properties.setProperty(org.springframework.integration.context.IntegrationProperties.THROW_EXCEPTION_ON_LATE_REPLY,
+					"" + integrationProperties.isThrowExceptionOnLateReply());
+			properties.setProperty(org.springframework.integration.context.IntegrationProperties.REQUIRE_COMPONENT_ANNOTATION,
+					"" + integrationProperties.isComponentAnnotationRequired());
+			return properties;
+		}
+
 	}
 
 	@Configuration
 	@ConditionalOnClass(EnableIntegrationMBeanExport.class)
 	@ConditionalOnMissingBean(value = IntegrationMBeanExporter.class, search = SearchStrategy.CURRENT)
 	@ConditionalOnProperty(prefix = "spring.jmx", name = "enabled", havingValue = "true", matchIfMissing = true)
-	@EnableIntegrationMBeanExport(defaultDomain = "${spring.jmx.default-domain:}", server = "${spring.jmx.server:mbeanServer}")
+	@EnableIntegrationMBeanExport(defaultDomain = "${spring.jmx.default-domain:}",
+			server = "${spring.jmx.server:mbeanServer}")
 	protected static class IntegrationJmxConfiguration {
 	}
 
+	@Configuration
+	@ConditionalOnWebApplication
+	@ConditionalOnClass(EnableIntegrationGraphController.class)
+	@ConditionalOnMissingBean(value = IntegrationGraphController.class, search = SearchStrategy.CURRENT)
+	@EnableIntegrationGraphController(path = "${spring.integration.graph-controller-path}")
```

Comment:
How is the `path` value handled? I don't think this will support relaxed binding, e.g. if someone sets `spring.integration.graphControllerPath` or `SPRING_INTEGRATION_GRAPH_CONTROLLER_PATH` the configuration will be ignored.